pkg/auth/providers/saml/saml_provider_test.go

```diff
@@ -0,0 +1,147 @@
+package saml
+
+import (
+	"context"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/auth/providers/ldap"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"testing"
+)
+
+func createMockProvider(t *testing.T, name string, hasLdap bool) *Provider {
+	// saml.Configure runs some ldap specific logic based on the saml provider name, so we provide
+	// just enough scaffolding to create the provider that way.
+	ctx := context.Background()
+	mgmtCtx, err := config.NewScaledContext(rest.Config{}, nil)
```

Comment:
Unfortunately, to use `config.NewScaledContext(rest.Config{}, nil)` and create a real scaled context you need a backing cluster to interact with, which we do not want to require in our unit tests. Currently, you have not tested any functions that use controllers, so you have not run into the problem. If you were to try something like `configuredProvider.getSamlConfig()`, you will always get an error because you never called `mgmtCtx.Start(...)`, which does not work without a live cluster to communicate with.

The alternative approach is to not use`Configure(...)`, which requires a mgmtCtx, and instead create a `Provider` struct with its fields populated with mocks. [gomock](https://github.com/golang/mock) makes mocking a lot easier, and if you need help getting started with it, I can provide some examples
```golang
	samlp := &Provider{
		ctx:         ctx,
		authConfigs: mockAuthConfigController,
		secrets:     mockSecretController,
		samlTokens:  mockSamlTokenController,
		userMGR:     mockUserManager,
		tokenMGR:    mockTokenManager,
		name:        name,
		userType:    name + "_user",
		groupType:   name + "_group",
	}
```

Reply:
Your current tests are good for testing the `Configure(...)` function, but you will need mocks to test any other functionality.